orchestration/hca_orchestration/solids/create_snapshot.py

```diff
@@ -72,6 +85,45 @@ def get_completed_snapshot_info(context: AbstractComputeExecutionContext, job_id
     yield Output(snapshot_info_dict['id'])
 
 
+@solid(
+    config_schema=Field(Permissive({"validate_snapshot_name": Field(bool, default_value=True, is_required=False)})),
+    required_resource_keys={'data_repo_client', 'snapshot_config', 'hca_manage_config', 'data_repo_service'},
+)
+def get_snapshot_from_project(context: AbstractComputeExecutionContext) -> Any:
+    """
+    Use the snapshot_name to get the associated snapshot_id from TDR,
+    for the snapshot created in the previous pipeline step - cut_snapshot.
+    If there is no matching snapshot or dataset - fail
+    If the snapshot_name does not end in the release_tag fail - so that we know we are using the correct snapshot
+    """
+    data_repo_service: DataRepoService = context.resources.data_repo_service
+    dataset_name = context.resources.snapshot_config.dataset_name
+    snapshot_name = context.resources.snapshot_config.snapshot_name
+    release_tag = context.resources.snapshot_config.qualifier
+    dataset = data_repo_service.find_dataset(dataset_name)
+
+    # we need the dataset to get the billing profile id, which is needed to query (enumerate) the snapshot
+    if not dataset:
+        raise Failure(f"Snapshot not found for dataset name [dataset_name={dataset_name}]")
+    if not snapshot_name:
+        raise Failure(f"Snapshot name not found for snapshot name [snapshot_name={snapshot_name}]")
```

Comment:
I don't think so. 
In order to use the release_tag to verify that we are making public the intended snapshot - I need the snapshot name rather than the dataset name, because the snapshot name has the release tag and the dataset name does not.
```
HCA Starting Make Snapshot Public
Snapshot name: hca_dev_003d56749bf64e51ab1b8fed80c308b9__20230728_20230807_dcp99
Snapshot ID: ed2f43fc-b4b2-4be1-aed9-b0c4f46afd78
Google Project ID: broad-jade-dev-data
Dataset: hca_dev_003d56749bf64e51ab1b8fed80c308b9__20230728_dcp2
```
Could we maybe Zoom or Huddle about this on Thursday? I'm not clear on why this will be falsey.

Reply:
I mean I guess I could just let it fail if the snapshot name doesn't end in release tag - if it can't find it, it doesn't end it it, so it fails...

```
if not snapshot_name.endswith(release_tag):
            raise Failure(f"Snapshot name does not end in current release tag [snapshot_name={snapshot_name}], \
            [release_tag={release_tag}].")
```

Reply:
Falsey in python strings usually means None or the empty string. There is a check on L108 for this condition with 
`if not snapshot_name:`

So it seems like when we get to L109 where the failure is raised, the code will emit a failure like ```Snapshot name not found for snapshot name [snapshot_name=None]```. 

The suggestion to check for the release tag seems reasonable.

Reply:
OH <lightbulb> I get it.
I'm using if not as a boolean check for existence - if it fails to find a snaphot_name it fails and the error says the name is "None" or something empty, rather than returning a snapshot name. yes?